### PR TITLE
new consoleoperator object handeling added to the console pkg + unit test

### PR DESCRIPTION
**Pull request by elenagerman**
new consoleoperator object handeling added to the console pkg + unit test

### CODE REVIEW

File: pkg/console/consoleoperator.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	}
+
+	if consoleOperatorName == "" {
+		glog.V(100).Info("The consoleOperatorName of the consoleOperator is empty")
+
+		return nil, fmt.Errorf("the consoleOperator 'consoleOperatorName' cannot be empty")
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("the consoleOperator object %s doesn't exist", consoleOperatorName)
```
Comment (sebrandon1) [inline pkg/console/consoleoperator.go:58]:
```suggestion
		return nil, fmt.Errorf("the consoleOperator object %s does not exist", consoleOperatorName)
```

File: pkg/console/consoleoperator.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+// GetPlugins fetches consoleOperator plugins list.
+func (builder *ConsoleOperatorBuilder) GetPlugins() (*[]string, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting consoleOperator plugins list configuration")
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("consoleOperator %s object doesn't exist", builder.Definition.Name)
```
Comment (sebrandon1) [inline pkg/console/consoleoperator.go:128]:
```suggestion
		return nil, fmt.Errorf("consoleOperator %s object does not exist", builder.Definition.Name)
```

File: pkg/console/consoleoperator_test.go [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+		{
+			name:                "",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("the consoleOperator 'consoleOperatorName' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                "consoletest",
+			addToRuntimeObjects: false,
+			expectedError:       fmt.Errorf("the consoleOperator object consoletest doesn't exist"),
```
Comment (sebrandon1) [inline pkg/console/consoleoperator_test.go:56]:
```suggestion
			expectedError:       fmt.Errorf("the consoleOperator object consoletest does not exist"),
```

**Review by sebrandon1 — APPROVED** [review]
Looks good, just a note about contractions per #385 and #390

File: pkg/console/console.go [hunk 4]
```diff
@@ -47,7 +47,7 @@ func NewBuilder(apiClient *clients.Settings, name string) *Builder {
 	return &builder
 }
 
-// Pull loads an existing console into the Builder struct.
+// Pull loads an existing console into the ConsoleOperatorBuilder struct.
```
Comment (trewest) [inline pkg/console/console.go:50]: I don't think you need this. The struct is still just called builder

File: pkg/console/console.go [hunk 5]
```diff
@@ -25,7 +25,7 @@ type Builder struct {
 	errorMsg string
 }
 
-// NewBuilder creates a new instance of Builder.
+// NewBuilder creates a new instance of ConsoleOperatorBuilder.
```
[the remark this answers is not in the document]
Comment (elenagerman) [reply pkg/console/console.go:28]: sure, it probably was change by mistake by refactoring

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**